test/any.js

```diff
@@ -56,4 +56,10 @@ describe('any', function() {
     });
   });
 
+  it('dispatches to objects that implement `reduce`', function() {
+    var obj = {x: [1, 2, 3], reduce: function() { return 'override'; }};
+    eq(R.any(odd, obj), 'override');
+    eq(R.any(R.equals(1), obj), 'override');
+  });
+
```

Comment:
I think we should use an example of a function that fulfills the [Foldable Law][fl].  The first thing that comes to mind is a simple wrapper around an array (adding little or no behavior.)  But a more powerful example might be a BTree structure with Branch and Leaf subtypes, so that we could write this:

```js
const tree = new Branch(
  new Branch(new Leaf('8'), '6', new Leaf('7')),
  '5',
  new Branch(new Leaf('3'), '0', new Leaf('9'))
)

any(n => n > 5, tree)  //-> true
any(n => n > 10, tree) //=> false
```
We'd have to write the Branch and Leaf classes, but they would be simply constructors with appropriate `reduce` functions, something like 

```js
(fn, init) => this.right.reduce(fn, fn(this.left.reduce(fn, init), this.val))
```

and

```js
(fn, init) => fn(init, this.val)
```

respectively.

  [fl]: https://github.com/fantasyland/fantasy-land#foldable